pymllm/engine/launch.py

```
@@ -0,0 +1,377 @@
+            audio_data=audio_data,
+            video_data=video_data,
+            lora_path=lora_path,
+            session_params=session_params,
+            extra_options=kwargs,
+        )
+        request.normalize_batch_and_arguments()
+
+        async def _run() -> Dict[str, Any]:
+            state = await self._rr_process.add_request(request)
+            if isinstance(rid, list):
+                raise ValueError("Synchronous `generate` currently supports single request.")
```

Ruff (0.15.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)